### ISSUE TITLE
[Bug] Stuff broken after updating to v3.4.3

### ISSUE THREAD

**Issue by tomasvitek** [ztalbot2000/homebridge-cmd4#104]
**Describe The Bug:**

Not sure what happened, but the latest update seems to have broken all my `cmd4` accessories. Something to do with the timeout stuff from the latest update. Sorry, don't have more time to investigate until next week. But figured I let you know about it anyway. Reverting to previous version fixes it. Below is an excerpt from my logs and configs. The error message repeats for all `cmd` accessories, and then it all repeats every 3-5 seconds.

**Logs:**
```
[4/9/2021, 2:28:42 AM] [Cmd4] getValue On function timed out undefinedms for light cmd: php /path/to/file.php Get 'light' 'On' Failed.
[4/9/2021, 2:28:42 AM] [Cmd4] getValue On function failed for light cmd: php /path/to/file.php Get 'light' 'On' Failed.  Error: null
```

**Cmd4 Config:**
```json
"platforms": [
        {
            "platform": "Cmd4",
            "name": "Cmd4",
            "outputConstants": false,
            "restartRecover": false,
            "Cmd4_Mode": "Always",
            "accessories": [
                {
                    "type": "Switch",
                    "displayName": "light",
                    "name": "Light",
                    "on": false,
                    "polling": true,
                    "Cmd4_Mode": "Always",
                    "stateChangeResponseTime": 1,
                    "state_cmd_prefix": "php",
                    "state_cmd": "/path/to/file.php"
                }
            ]
        },

```

**Screenshots:**

**Environment:**

* **Node.js Version**: v14.16.0
* **NPM Version**: v6.14.11
* **Homebridge Version**: v1.3.4
* **homebridge-cmd4 Version**: v3.4.3
* **Operating System**: docker on debian 
* **Process Supervisor**: docker

Thanks and take care! 🙂

**Comment by tomasvitek** [comment]
Btw I asked in my previous bug report (hi again!), but you probably missed it -- is there anywhere I can tip you for a coffee maybe?

**Comment by ztalbot2000** [comment]
That is really odd as if there is no timeout, the default is used.

To quickly fix this, add
"timeout": 6000

Inside your config, after polling for example.  Hmm, very odd.

Ttyl,
John

**Comment by ztalbot2000** [comment]
This was the function added,


Let timeout = constants.DEFAULT_TIMEOUT;

      // Secondly the accessories timeout
      if ( accessory.timeout )
         timeout = accessory.timeout;

      // Finally the polled timeout setting
      let pollingEntrys = settings.arrayOfPollingCharacteristics.filter(
           entry => entry.accessory.UUID == accessory.UUID &&
           entry.accTypeEnumIndex == accTypeEnumIndex );

      // There should only be one, if any
      if ( pollingEntrys.length > 0 )
         timeout = pollingEntrys[0].timeout;

      return timeout;


I don't quite understand how this could be, but will investigate,
John

**Comment by tomasvitek** [comment]
> To quickly fix this, add "timeout": 6000

Thanks for the quick reply. I did as you suggested. Added it all accessories after `polling`. Surprisingly, no change. Still the same errors.

**Comment by tomasvitek** [comment]
So it looks like it might get set, just maybe gets lost on the way somewhere? 🤔 Anyway, sorry have to go now, and will not be available until Sunday. But happy to help debug then. Have a nice weekend!

**Comment by ztalbot2000** [comment]
That is really so strange.

Try:

"polling":{ "characteristic": "On", "timeout": 6000}

**Comment by ztalbot2000** [comment]
Oops,

"Polling":[{"characteristic": "On", "timeout": 6000}],

It's an array

**Comment by tomasvitek** [comment]
This is very strange then. I tried the first option you suggested:

```
"polling":{ "characteristic": "On", "timeout": 6000}
```

and it worked! 🤷🏼‍♂️ No errors, I seem to be getting up to date state and setting also works.

Btw is there any functional difference between this and `polling: true`?

**Comment by tomasvitek** [comment]
The array works as well, btw.

**Comment by ztalbot2000** [comment]
Polling:true uses the default polling value for the On characteristic.
Probably 6000ms

I am recreating the issue now.

Ttyl,
John

**Comment by ztalbot2000** [comment]
Just published v3.4.4,  Sorry about that.  One day one bug fixed, only to
reveal another.  Should have left them alone I guess.

Ttyl,
John

**Comment by tomasvitek** [comment]
John, thank you for the quick fix! Have a nice weekend!
